runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/util/common/worker/MapTaskExecutorTest.java

```diff
@@ -483,4 +483,29 @@ public void testExceptionInAbortSuppressed() throws Exception {
       assertThat(e.getSuppressed()[0].getMessage(), equalTo("suppressed in abort"));
     }
   }
+
+  @Test
+  public void testAbort() throws Exception {
+    // Operation must be an instance of ReadOperation or ReceivingOperation per preconditions
+    // in MapTaskExecutor.
+    Operation o = Mockito.mock(ReadOperation.class);
+
+    ExecutionStateTracker stateTracker = ExecutionStateTracker.newForTest();
+    try (MapTaskExecutor executor =
+        new MapTaskExecutor(Arrays.<Operation>asList(o), counterSet, stateTracker)) {
+      Mockito.doAnswer(
+              invocation -> {
+                executor.abort();
+                return null;
+              })
+          .when(o)
+          .start();
+      executor.execute();
+      fail("Should have aborted");
+    } catch (Exception e) {
+      Assert.assertTrue(e instanceof InterruptedException);
```

Comment:
It would be nice for the next person if there is a diagnostic error message attached to this failure, lest we get a failure that says "false is not true".

Reply:
Done.